src/ArrayOfArraysView.hpp

```diff
@@ -734,6 +734,40 @@ class ArrayOfArraysView
     }, m_values, buffers ... );
   }
 
+  /**
+   * @brief Clears the array and creates a new array with the given number of sub-arrays.
+   * @param numSubArrays The new number of arrays.
+   * @param offsets A pointer to an array of length @p numSubArrays+1 containing the offset
+   *   of each new sub array. Offsets are precomputed by the caller.
+   * @param buffers A variadic pack of buffers to treat similarly to m_values.
+   */
+  template< typename ... BUFFERS >
+  void resizeFromOffsets( INDEX_TYPE const numSubArrays,
+                          INDEX_TYPE const * const offsets,
+                          BUFFERS & ... buffers )
```

Comment:
I still need to add a unit test for it. But I'd like to get the idea approved first. Usage is as follows:
```c++
newAofA.resizeFromOffsets( otherAofA.size(), otherAofA.toView().getOffsets() );
```

Reply:
Yeah now that I think about it (in the context of a `CRSMatrix` at least) `duplicateSparsity` would have a different meaning. You'd basically copy all the data wholesale, except set the new values to zero. `resizeFromOffsets` seems like the correct name, although I'm not a huge fan of passing the pointer and size around, but it's not a big deal.